openapi-to-har.js

```diff
@@ -104,21 +104,43 @@ const getPayload = function (openApi, path, method) {
   }
 
   if (
-    openApi.paths[path][method].requestBody &&
-    openApi.paths[path][method].requestBody.content &&
-    openApi.paths[path][method].requestBody.content['application/json'] &&
-    openApi.paths[path][method].requestBody.content['application/json'].schema
-  ) {
-    const sample = OpenAPISampler.sample(
-      openApi.paths[path][method].requestBody.content['application/json']
-        .schema,
-      { skipReadOnly: true },
-      openApi
-    );
-    return {
-      mimeType: 'application/json',
-      text: JSON.stringify(sample),
-    };
+     openApi.paths[path][method].requestBody &&
+     openApi.paths[path][method].requestBody.content
+   ) {
+     if (openApi.paths[path][method].requestBody.content['application/json'] &&
+       openApi.paths[path][method].requestBody.content['application/json'].schema
+     ) {
+       const sample = OpenAPISampler.sample(
+         openApi.paths[path][method].requestBody.content['application/json']
+           .schema,
+         { skipReadOnly: true },
+         openApi
+       );
+       return {
+         mimeType: 'application/json',
+         text: JSON.stringify(sample)
+       };
+     }
+
+     if (openApi.paths[path][method].requestBody.content['multipart/form-data'] &&
+       openApi.paths[path][method].requestBody.content['multipart/form-data'].schema
+     ) {
+       const sample = OpenAPISampler.sample(openApi.paths[path][method].requestBody.content['multipart/form-data']
+           .schema,
+         {skipReadOnly: true},
+         openApi
+       );
+
+       if (sample === undefined) return null;
+
+       const params = [];
+        Object.keys(sample).forEach(key => params.push({'name': key, 'value': sample[key]}));
+
+       return {
+         mimeType: 'multipart/form-data',
+         params: params,
+       };
+     }
```

Comment:
I took a stab at DRYing up the duplication. It's probably fine as-is and I wasn't able to cut it down by much, but use this if you think it's clearer.

```javascript
['application/json', 'multipart/form'].forEach((type) => {
  content = openApi.paths[path][method].requestBody.content[type]
  if (content && content.schema) {
    const sample = OpenAPISampler.sample(
      content.schema,
      { skipReadOnly: true },
      openApi
    );
    if (type === 'application/json') {
      return {
        mimeType: type,
        text: JSON.stringify(sample)
      };
    } else if (type === 'multipart/form-data') {
      let params;
      if (sample !== undefined) {
        params = Object.keys(sample).forEach(key => params.push({'name': key, 'value': sample[key]}));
      }
      return {
        mimeType: type,
        params: params
      };
    }
  }
});

```

Reply:
I'll address this in #3 since it relates there more.

Reply:
addressed here: https://github.com/bpross/openapi-snippet/pull/3/commits/66b0dba4f60fcc0fa0cdc4fc09aab4f9604aabf5